.gdbinit

```diff
@@ -336,27 +386,35 @@ define ____print_ht
 			set $h = $i
 		else
 			set $bucket = (Bucket*)($ht->arData + $i)
-			set $val = &$bucket->val
+			set $val = &($bucket->val)
 			set $key = $bucket->key
 			set $h = $bucket->h
 		end
+
 		set $n = $ind
 		if $val->u1.v.type > 0
 			while $n > 0
 				printf "  "
 				set $n = $n - 1
 			end
 			printf "[%d] ", $i
+
+			# print hash key
 			if $key
+				# for bucket, $key != 0 and is zend_string*;
 				____print_str $key->val $key->len
 				printf " => "
 			else
+				# for packed $key == 0
 				printf "%d => ", $h
 			end
+
+			# print hash value
```

Comment:
I believe this just prints the zval pointer instead of more details, hash sounds like the string hash value.

Reply:
Yes. You're right. I will make it clear.